### ISSUE TITLE
DiscordChannel.GetMessageAsync returns no Reactions when command is running from a replied message

### ISSUE THREAD

**Issue by Olimp666** [DSharpPlus/DSharpPlus#792]
I'm trying to get the reactions of the message.
```        
        [Command("reactions")]
        public async Task ReactionsCommand(CommandContext ctx)
        {
            var channel = ctx.Channel;
            var emoji =await channel.GetMessageAsync(817299244008210443);
        }
```
![image](https://user-images.githubusercontent.com/47377233/110101020-6c6d8900-7de6-11eb-9136-7c9e0de07484.png)
![image](https://user-images.githubusercontent.com/47377233/110101082-81e2b300-7de6-11eb-9f66-166ddd019354.png)

It works fine unless i reply to some other message with this command.
![image](https://user-images.githubusercontent.com/47377233/110100811-2dd7ce80-7de6-11eb-9b75-1a4fcda1811f.png)
![image](https://user-images.githubusercontent.com/47377233/110100893-49db7000-7de6-11eb-8069-03537525efc2.png)

Basically, I want to get the reactions of the replied message, but when i reply with "+reactions", DiscordChannel.GetMessageAsync returns everything except Reactions.

**Comment by Neuheit** [comment]
Are you able to provide a usable paste rather than a hardcoded message ID?

**Comment by Olimp666** [comment]
Sure
Using this code the msg.Reactions will be "Count=0"
```
[Command("reactions")]
public async Task ReactionsCommand(CommandContext ctx)
{
    var msg = await ctx.Channel.GetMessageAsync(ctx.Message.ReferencedMessage.Id);
}
```

**Comment by Neuheit** [comment]
Unfortunately it looks like this is a Discord limitation. D#+ currently uses a cache from the gateway to retrieve both that message and the message being replied, and it doesn't look like the gateway sends a reaction collection. With that being said it looks like this reaction collection is only sent via REST. In the future you will have the ability to specify the lib to force a rest request rather than going through the cache, but until then you'll have to use the `DiscordRestClient` to grab that information exclusively from REST. 

Hope this makes sense.

**Comment by Neuheit** [comment]
You can view these raw payloads from Discord by setting your log level to trace.

**Comment by Olimp666** [comment]
Ok i fugured out how to get the reactions but it's a bit spaghetti

```
            var client = new DiscordClient(new DiscordConfiguration()
            {
                Token = "ODEwMzEwNjk4MTEwMDI1NzMw.YChypQ.td4gwTmdYKfIET6_c9cvV3fmCcc",
                Intents = DiscordIntents.All,
                MinimumLogLevel = Microsoft.Extensions.Logging.LogLevel.Information
            });
            var ch = await client.GetChannelAsync(ctx.Channel.Id);
            
            var msg = await ch.GetMessageAsync(ctx.Message.ReferencedMessage.Id);
```

**Comment by Olimp666** [comment]
var client = new DiscordClient(new DiscordConfiguration()
            {
                Token = "token",
                Intents = DiscordIntents.All,
                MinimumLogLevel = Microsoft.Extensions.Logging.LogLevel.Information
            });
            var ch = await client.GetChannelAsync(ctx.Channel.Id);
            
            var msg = await ch.GetMessageAsync(ctx.Message.ReferencedMessage.Id);

**Comment by Olimp666** [comment]
Ok I figured out how to get the reactions but it's a bit spaghetti

            var client = new DiscordClient(new DiscordConfiguration()
            {
                Token = "token",
                Intents = DiscordIntents.All,
                MinimumLogLevel = Microsoft.Extensions.Logging.LogLevel.Information
            });
            var ch = await client.GetChannelAsync(ctx.Channel.Id);
            
            var msg = await ch.GetMessageAsync(ctx.Message.ReferencedMessage.Id);
